### PR TITLE
feat: Added manifest generation to snowflake to s3 task.

### PR DESCRIPTION
**Pull request by HassanJaveed84**
feat: Added ability to load data from s3 to mysql using a manifest file

### CODE REVIEW

File: edx_prefectutils/s3.py [hunk 1]
```diff
@@ -8,6 +8,26 @@
 from prefect.utilities.aws import get_boto_client
 
 
+@task
+def delete_s3_directory(bucket: str = None, prefix: str = None, credentials: dict = None):
```
Comment (estute) [inline edx_prefectutils/s3.py:12]: nit on the naming of this: does it actually delete the directory, or will it leave an empty prefix?

Comment (HassanJaveed84) [reply edx_prefectutils/s3.py:12]: If you delete all the keys under a prefix, the prefix is removed as well by S3.

File: edx_prefectutils/snowflake.py [hunk 2]
```diff
@@ -461,6 +470,16 @@ def export_snowflake_table_to_s3(
     table_name = qualified_table_name(sf_database, sf_schema, sf_table)
     export_path = os.path.join(s3_path, table_name.replace('.', '-').lower()) + '/'
 
+    parsed_path = urlparse(export_path)
+    export_bucket = parsed_path.netloc
+    export_prefix = parsed_path.path.lstrip('/')
+
+    # Snowlflake's COPY INTO command's OVERWRITE option is not reliable, so we need to delete the data manually
```
Comment (brianhw) [inline edx_prefectutils/snowflake.py:477]: nit:  Snowlflake => Snowflake... : )

File: edx_prefectutils/s3.py (hunk 1 above)
Comment (estute) [reply edx_prefectutils/s3.py:12]: thanks.

File: edx_prefectutils/snowflake.py [hunk 3]
```diff
@@ -461,6 +470,16 @@ def export_snowflake_table_to_s3(
     table_name = qualified_table_name(sf_database, sf_schema, sf_table)
     export_path = os.path.join(s3_path, table_name.replace('.', '-').lower()) + '/'
 
+    parsed_path = urlparse(export_path)
+    export_bucket = parsed_path.netloc
+    export_prefix = parsed_path.path.lstrip('/')
+
+    # Snowflake's COPY INTO command's OVERWRITE option is not reliable, so we need to delete the data manually
```
Comment (estute) [inline edx_prefectutils/snowflake.py:477]: I have a question about how `OVERWRITE` works. Assume you have files A and B in your bucket and you go to run a copy into command, with OVERWRITE, but only operating with file A. Is the expectation that Snowflake's COPY INTO command would result in a bucket containing only A, or merely overwrite file A, leaving file B intact?

File: tests/test_mysql.py [hunk 4]
```diff
@@ -170,3 +170,26 @@ def test_table_creation_with_indexes(mock_mysql_connection):
             mock.call("\n        CREATE TABLE IF NOT EXISTS test_table (user_id int,course_id varchar(255) NOT NULL,INDEX (user_id),INDEX (course_id),INDEX (user_id,course_id))\n    "), # noqa
         ]
     )
+
+
+def test_load_s3_data_to_mysql_with_manifesat(mock_mysql_connection):
```
Comment (estute) [inline tests/test_mysql.py:175]: nit: manifesat/manifest

**Review by estute — APPROVED** [review]
LGTM

File: edx_prefectutils/snowflake.py (hunk 3 above)
Comment (HassanJaveed84) [reply edx_prefectutils/snowflake.py:477]: The COPY command will only remove files that match the files that the COPY command unloads. 
I've seen cases where a previous run of COPY command generated files `A` and `B`, and a future run with overwrite generates files `A` and `C`, this future run would overwrite files `A` and `C` but would leave `B`, which is not what we want as they might get loaded into aurora downstream of this.

Comment (HassanJaveed84) [reply edx_prefectutils/snowflake.py:477]: That is why I also added the manifest generation and load from manifest option to have better control on what gets loaded into MySQL.

**Pull request merged**